tracee-ebpf/tracee/events.go

```diff
@@ -0,0 +1,184 @@
+package tracee
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"strconv"
+	"unsafe"
+
+	"github.com/aquasecurity/tracee/tracee-ebpf/external"
+)
+
+type RawEvent struct {
+	Ctx      context
+	Args     map[string]interface{}
+	ArgMetas []external.ArgMeta
+}
+
+// context struct contains common metadata that is collected for all types of events
+// it is used to unmarshal binary data and therefore should match (bit by bit) to the `context_t` struct in the ebpf code.
+// NOTE: Integers want to be aligned in memory, so if changing the format of this struct
+// keep the 1-byte 'Argnum' as the final parameter before the padding (if padding is needed).
+type context struct {
+	Ts       uint64
+	Pid      uint32
+	Tid      uint32
+	Ppid     uint32
+	HostPid  uint32
+	HostTid  uint32
+	HostPpid uint32
+	Uid      uint32
+	MntID    uint32
+	PidID    uint32
+	Comm     [16]byte
+	UtsName  [16]byte
+	ContID   [16]byte
+	EventID  int32
+	Retval   int64
+	StackID  uint32
+	Argnum   uint8
+	_        [3]byte //padding
+}
+
+func (t *Tracee) processEvents(done <-chan struct{}) error {
+	for dataRaw := range t.eventsChannel {
+		dataBuff := bytes.NewBuffer(dataRaw)
+		var ctx context
+		err := binary.Read(dataBuff, binary.LittleEndian, &ctx)
+		if err != nil {
+			t.handleError(err)
+			continue
+		}
+
+		rawEvent := RawEvent{
+			Ctx:      ctx,
+			Args:     make(map[string]interface{}, ctx.Argnum),
+			ArgMetas: make([]external.ArgMeta, ctx.Argnum),
+		}
+
+		params := EventsIDToParams[ctx.EventID]
```

Comment:
so sometimes we're using ctx directly, sometimes using the rawevent type... like in line #87

Reply:
As I explained above, creating this type is necessary to avoid arguments iteration, which also simplifies the code in:
1. `shouldProcessEvent`
2. `processEvent`
3. `prepareArgs`

Have a look, for example, in `prepareArgs` where you can clearly see the advantage of using an arguments map

Reply:
We shouldn't, removed, thanks!